Added ability to ignore fields by tagging them.

mikijov opened the pull request:
Added code and tests to support tagging fields with testdiff:"ignore".
Adding this tag to any field will make messagediff skip this field.
This is useful for transient fields that do not store actual value of
the struct. I have used generic 'testdiff' tag to potentially make it
usable by other libraries who might want to do the same purpose.

d4l3k commented on README.md:
```diff
@@ -17,11 +20,12 @@ import "gopkg.in/d4l3k/messagediff.v1"
-			a := someStruct{1, 2, []int{1}}
-			b := someStruct{1, 3, []int{1, 2}}
+			a := someStruct{1, 2, []int{1}, 9}
+			b := someStruct{1, 3, []int{1, 2}, 10}
```
Probably should add a separate example to show off this feature since most people won't care about it.

mikijov replied:
Do you mean to create a new test case, or to create a new Test with it's own struct?

mikijov replied:
OK, I separated the test for ignoring fields into a separate test altogether.

mikijov commented:
I mistakenly thought your review comment related to actual tests and not README... I fixed it in docs now. I also corrected indentation of the code in README.
